### ISSUE TITLE
Add units to rubix class

### ISSUE THREAD

**Issue by anschaible** [ufuk-cakir/rubix#111]
Add units to the rubix class and change the print statement

**Comment by ufuk-cakir** [comment]
how are you planning of adding units to the rubix class? the first time I tried doing this it was a bit challenging since the arrays inside the class need to be JAX arrays such that JAX can trace them and can compute the gradient. The [jpu package](https://github.com/dfm/jpu) might be interesting to look at

**Comment by anschaible** [comment]
See branch 11 add units to rubix class

jax cannot directly handle the units, but we add the units to the rubix object just before returning rubixdata from the pipeline

**Comment by TobiBu** [comment]
> The [jpu package](https://github.com/dfm/jpu) might be interesting to look at

yeah, jau would ultimately be nice, but since its still rough and in development, for now we have just modified the representation of RubixData class so that the printing looks nicer formatted. 
We do not yet support unit conversion internally. I would suggest we do keep an eye on [jpu](https://github.com/dfm/jpu) and keep our first approach until then.
